libcron/include/libcron/Task.h

```diff
@@ -1,26 +1,56 @@
 #pragma once
 
 #include <functional>
-#include "CronData.h"
-#include "CronSchedule.h"
 #include <chrono>
 #include <utility>
+#include "CronData.h"
+#include "CronSchedule.h"
 
 namespace libcron
 {
-    class Task
+    class TaskInformation
+    {
+        public:
+            virtual ~TaskInformation() {}
+            virtual std::chrono::system_clock::duration get_delay() const = 0;
+    };
+
+    class Task : public TaskInformation
     {
         public:
+            using TaskFunction = std::function<void(const TaskInformation*)>;
 
-            Task(std::string name, const CronSchedule schedule, std::function<void()> task)
+            class TaskProxy
+            {
+                public:
+                TaskProxy(TaskFunction task) : task(std::move(task)) {}
+
+                void operator() (const TaskInformation* i)
+                {
+                    task(i);
+                }
+
+                private:
+                TaskFunction task;
+            };
```

Comment:
Can't we remove this class now that we're no longer supporting dual callback signatures?

Reply:
If we want to ensure full lambda-support in passing callbacks, I am afraid no.

Reply:
But we're not, are we? The new interface is for future proofing the API. Right? Or am I just too tired to do code review?

Reply:
Ah, you are right, we can get rid of the proxy as long as we template `add_schedule`! Then it works as expected.

Reply:
Done!

Reply:
Why did you have to add a third template argument to add_schedule? Was it because `TaskFunction` wasn't available to you? If so, lets move that using alias instead such that it is available where it is needed instead.

Reply:
Yeah, you are right, somehow this was related to the missing auto& issue I had and then I have forgot to clean this up as well... Now everything should be done!